include/nbl/asset/utils/IMeshManipulator.h

```diff
@@ -451,11 +457,32 @@ class IMeshManipulator : public virtual core::IReferenceCounted
 		}
 
 		//! Recalculates the cached bounding box of the meshbuffer
-		static inline void recalculateBoundingBox(ICPUMeshBuffer* meshbuffer)
+		static inline void recalculateBoundingBox(ICPUMeshBuffer* meshbuffer, bool isOBBDisabled)
 		{
-			meshbuffer->setBoundingBox(calculateBoundingBox(meshbuffer,meshbuffer->getJointAABBs()));
+      if(isOBBDisabled)
+      { meshbuffer->setBoundingBox(calculateBoundingBox(meshbuffer,meshbuffer->getJointAABBs())); }
+      else
+      { meshbuffer->setBoundingBox(calculateOrientedBBox(meshbuffer)); }
 		}
 
+    static inline core::OBB calculateOrientedBBox(const ICPUMeshBuffer* meshBuffer)
+    {
+      auto vtxCount = meshBuffer->getIndexCount();
```

Comment:
this is wrong, Index Count != Vertex Count

Reply:
I am sure it gives me exact number of vertices on a mesh, checked out the math already!

Reply:
the other method calcVertexCount is actually wrong which max out at 16 whatsoever!

Reply:
`upperBoundVertexID` method seems to be doing vertex count calculation. It simply can be replaced, though weird naming for the method. Eitherway, there's no reason to rule out other things like dynamic allocation of vertices as well as looping through the indices (either by vtx count or index count) to get the position vector claiming it triggers out of bounds assert as it doesn't in any case. The only problem remains the duplicate of vertices which simply can be avoided by correct calculation, but in any case, the final calculation is not affected by any of these factors as they will be overwritten in the iterations down the line as there's a constraint to which they iterated by, which is the projection count.